scipy/cluster/hierarchy.py

```diff
@@ -3555,7 +3555,7 @@ def _dendrogram_calculate_info(Z, p, truncate_mode,
         nb = 1
         db = 0.0
 
-    if count_sort == 'ascending' or count_sort:
+    if count_sort == 'ascending' or count_sort is True:
```

Comment:
This bug was introduced in gh-7432. Looking at the history, it used to be `count_sort == True` which is slightly more permissive than `is True`, allowing numpy booleans for example. I also see that `distance_sort` on line 3579 was broken in the same way, could you fix that as well?

It would also be good to add a test for `count_sort='descending'` to catch errors like this in future.

Reply:
@peterbell10, thanks for finding that commit.  Using `==` instead of `is` makes sense.  In the same commit, a few lines down, the `if` statement with `distance_sort` got the same treatment, so my guess is that change should also be reverted.